mungegithub/mungers/flake-manager.go

```diff
@@ -0,0 +1,206 @@
+	}
+	p.oldestTime = time.Now().Add(-time.Hour * 24)
+	p.alreadySyncedFlakes = map[cache.Flake]int{}
+	p.config = config
+	p.googleGCSBucketUtils = utils.NewUtils(URLTestStorageBucket)
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (p *FlakeManager) EachLoop() error {
+	if p.sq.e2e == nil {
+		return fmt.Errorf("submit queue not initialized yet")
+	}
+	if !p.finder.Synced() {
+		return nil
+	}
+	p.sq.e2e.GCSBasedStable()
+	for _, f := range p.sq.e2e.Flakes() {
+		p.syncFlake(f)
+	}
+	return nil
+}
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (p *FlakeManager) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// Munge is unused by this munger.
+func (p *FlakeManager) Munge(obj *github.MungeObject) {}
+
+func (p *FlakeManager) syncFlake(f cache.Flake) error {
+	if _, ok := p.alreadySyncedFlakes[f]; ok {
+		return nil
+	}
+	if n, ok := p.finder.IssueForKey(string(f.Test)); ok {
+		if err := p.updateIssue(n, f); err != nil {
+			return fmt.Errorf("error updating issue %v for %v: %v", n, f, err)
+		}
+		p.alreadySyncedFlakes[f] = n
+		return nil
+	}
+	n, err := p.createIssue(f)
+	if err != nil {
+		return fmt.Errorf("error making issue for %v: %v", f, err)
+	}
+	p.finder.Created(string(f.Test), n)
+	p.alreadySyncedFlakes[f] = n
+	return nil
+}
+
+// DO NOT CHANGE or it will not recognize previous entries!
+func (p *FlakeManager) flakeID(flake cache.Flake) string {
+	return p.googleGCSBucketUtils.GetPathToJenkinsGoogleBucket(string(flake.Job), int(flake.Number), "") + "\n"
+}
+
+func (p *FlakeManager) flakeExtraInfo(flake cache.Flake) string {
+	return fmt.Sprintf("Failed: %v\n\n```\n%v\n```\n\n", flake.Test, flake.Reason)
+}
+
+func (p *FlakeManager) updateIssue(issueNumber int, flake cache.Flake) error {
+	flakeID := p.flakeID(flake)
+	obj, err := p.config.GetObject(issueNumber)
+	if err != nil {
+		return fmt.Errorf("error getting object for %v: %v", issueNumber, err)
+	}
+	comments, err := obj.ListComments()
+	if err != nil {
+		return fmt.Errorf("error getting comments for %v: %v", *obj.Issue.Number, err)
+	}
+	if obj.Issue.Body != nil && strings.Contains(*obj.Issue.Body, flakeID) {
+		// We already wrote this flake
+		return nil
+	}
+	for _, c := range comments {
+		if c.Body == nil {
+			continue
+		}
+		if strings.Contains(*c.Body, flakeID) {
+			// We already wrote this flake
+			return nil
+		}
+	}
+	glog.Infof("Updating issue %v with flake %v", issueNumber, flake)
+	return obj.WriteComment(flakeID + "\n" + p.flakeExtraInfo(flake))
+}
+
+func (p *FlakeManager) createIssue(flake cache.Flake) (issueNumber int, err error) {
+	obj, err := p.config.NewIssue(
+		string(flake.Test),                            // title
+		p.flakeID(flake)+"\n"+p.flakeExtraInfo(flake), // body
+		[]string{"kind/flake"},                        // labels
+	)
+	if err != nil {
+		return 0, err
+	}
+	glog.Infof("Created issue %v for flake %v", *obj.Issue.Number, flake)
+	return *obj.Issue.Number, nil
+}
+
+/*
```

Comment:
whoops, will do

On Mon, May 23, 2016 at 1:20 PM, Mike Danese <notifications@github.com>
wrote:

> In mungegithub/mungers/flake-manager.go
> <https://github.com/kubernetes/contrib/pull/1031#discussion_r64281700>:
>
> > +}
> > +
> > +func (p *FlakeManager) createIssue(flake cache.Flake) (issueNumber int, err error) {
> > +	obj, err := p.config.NewIssue(
> > +		string(flake.Test),                            // title
> > +		p.flakeID(flake)+"\n"+p.flakeExtraInfo(flake), // body
> > +		[]string{"kind/flake"},                        // labels
> > +	)
> > +	if err != nil {
> > +		return 0, err
> > +	}
> > +	glog.Infof("Created issue %v for flake %v", *obj.Issue.Number, flake)
> > +	return *obj.Issue.Number, nil
> > +}
> > +
> > +/*
>
> remove dead code?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/contrib/pull/1031/files/f59088528710f2bb7e698363ace01206a934180a#r64281700>
>